pandas/_libs/index.pyx

```diff
@@ -534,7 +534,9 @@ cpdef convert_scalar(ndarray arr, object value):
         elif isinstance(value, (datetime, np.datetime64, date)):
             return Timestamp(value).value
         elif value is None or value != value:
-            return NPY_NAT
+            if not util.is_timedelta64_object(value):
```

Comment:
can we simply check is_timedelta64_object at a higher level (e.g. the if/elif level) to avoid the nested if calls (here and below), and simply dispatch on it directly

Reply:
sure